### ISSUE TITLE
[tune] PBT Model accuracy inconsistent with training

### ISSUE THREAD

**Issue by Nithanaroy** [ray-project/ray#4538]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux
- **Ray installed from (source or binary)**: binary
- **Ray version**: 0.6.5
- **Python version**: 2.7
- **Exact command to reproduce**: NA

### Describe the problem

I used tune's PBT for hyperparameter discovery. The training accuracy given by `model.evaluate()` of the final model saved by `_stop()` of `Trainable` instance on the same dataset, is off by **15%** with what is shown on Tensorboard. What am I missing?

### Source code / logs

```python
class MyTrainable(Trainable):
...
   def _train(self):
            h = self.model.fit(
                    self.train_data,
                    steps_per_epoch=...,
                    epochs=1,
                    validation_data=self.test_data,
                    validation_steps=...,
                    verbose=0,
                    callbacks=[]
                )
            metrics = dict([(metric, h.history[metric][-1]) for metric in h.history])
            metrics["neg_val_loss"] = -metrics["val_loss"]
        
           return metrics

    def _stop(self):
        saved_path = os.path.join(self.logdir, "model-fin.h5")
        self.model.save(saved_path)
        print("Saved model at %s" % saved_path)
```
**Train Spec:**
```python
train_spec = {
    "run": MyTrainable,
    "resources_per_trial": {
        "cpu": 4,
        "gpu": 1
    },
    "local_dir": "./ray_results",
    "max_failures": 2,
    "stop": {
        "neg_val_loss": -0.3,
        "training_iteration": 600
    },
    "config": {
        # decides how each of these params start off - can be fixed or randomly drawn from a distribution
        "decay": sample_from( lambda spec: spec.config.lr / Y2.0 ),
        "lr": sample_from( lambda _: 10 ** np.random.uniform(-5, -3) ),
        "layer1": sample_from( lambda _: np.random.randint(X11, Y11) ),
        "layer2": sample_from( lambda _: np.random.randint(X12, Y12) ),
        "layer3": sample_from( lambda _: np.random.randint(X13, Y13) ),
        "layer4": sample_from( lambda _: np.random.randint(X14, Y14) ),
        "layer5": sample_from( lambda _: np.random.randint(X15, Y15) ),
        "layer6": sample_from( lambda _: np.random.randint(X21, Y21) ),
        "layer7": sample_from( lambda _: np.random.randint(X22, Y21) ),
        "layer8": sample_from( lambda _: np.random.randint(X23, Y23) ),
        "layer9": sample_from( lambda _: np.random.randint(X24, Y24) ),
        "layer10": sample_from( lambda _: np.random.randint(X25, Y25) )
    },
    "num_samples": 20,
}
```
**Initialize PBT**
```python
pbt = PopulationBasedTraining(
        time_attr="training_iteration",
        reward_attr="neg_val_loss",
        perturbation_interval=10,
        hyperparam_mutations={
            "lr": lambda: 10 ** np.random.uniform(-5, -3),
    })
```
Here `tf.keras.model` is used.

Is there any problem with `hyperparam_mutations` or `perturbation_interval` or `reward_attr` or train spec config? I use the path printed in the `stop()` method while training to run `model_evaluate` on. Or do I have to wait till PBT stops to get the final model?

**Comment by krfricke** [comment]
Hi @Nithanaroy, I know it's been over 1.5 years, sorry for the large delay. You probably solved this issue by now or moved on.

For anyone finding this issue, the main thing I'd be looking at would be if you are using the same evaluation data/dataset as during training. If this is a test dataset and tensorboard reports accuracy on the validation dataset, differences are to be expected.

I'm closing this issue, feel free to re-open or open a new issue if this problem remains.
